ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -65,6 +67,8 @@ public DefaultLocalPersistenceService(final DefaultPersistenceConfiguration pers
       throw new NullPointerException("DefaultPersistenceConfiguration cannot be null");
     }
     lockFile = new File(rootDirectory, ".lock");
+    cleanShutdown = new File(rootDirectory, ".cleanShutdown");
```

Comment:
Trying to understand the need for two files here:

| &darr; unclean.exists() / clean.exists() &rarr; | true | false |
|-|-|-|
|true| clean (no-op) | unclean (delete) |
|false| clean (no-op) | clean (no-op) |

Since there are only two output states... why not just have a single "clean" marker? Look for it on startup (and then delete it), and then create it on shutdown.